frontends/common/resolveReferences/resolveReferences.cpp

```diff
@@ -213,24 +214,20 @@ void ResolveReferences::resolvePath(const IR::Path* path, bool isType) const {
     refMap->setDeclaration(path, decl);
 }
 
-void ResolveReferences::checkShadowing(const IR::INamespace*ns) const {
+void ResolveReferences::checkShadowing(const IR::INamespace* ns) const {
     if (!checkShadow) return;
-
-    auto e = ns->getDeclarations();
-    while (e->moveNext()) {
-        const IR::IDeclaration* decl = e->getCurrent();
+    for (auto decl : *ns->getDeclarations()) {
         const IR::Node* node = decl->getNode();
         auto prev = context->resolve(decl->getName(), ResolutionType::Any, !anyOrder);
         if (prev->empty()) continue;
+
         for (auto p : *prev) {
             const IR::Node* pnode = p->getNode();
             if (pnode == node) continue;
-            if (pnode->is<IR::Type_Method>() && node->is<IR::Type_Method>()) {
-                auto md = node->to<IR::Type_Method>();
-                auto mp = pnode->to<IR::Type_Method>();
-                if (md->parameters->size() != mp->parameters->size())
-                    continue;
-            }
+            if (pnode->is<IR::Type_Method>() && node->is<IR::Type_Method>())
+                // Methods can overload each other if they have a different number of arguments
```

Comment:
It will just avoid giving a shadowing warnings for all methods.
There is another check somewhere that two methods may not have the same signature.